### PR TITLE
ScyllaDB Data Connector

### PR DESCRIPTION
**Pull request by lukekim**
This pull request adds support for ScyllaDB as a federated data source in the project. The changes introduce new modules, feature flags, and configuration to enable integration with ScyllaDB, including connection pooling, table provider factories, and test coverage. The implementation includes updates to several core crates and a new example Spicepod configuration for federated ScyllaDB datasets.

**ScyllaDB Integration**

* Added `scylla` as a dependency and feature flag in `Cargo.toml` files for the main workspace and key crates (`data_components`, `db_connection_pool`, `runtime`). [[1]](diffhunk://#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R245) [[2]](diffhunk://#diff-bcbe44138259c7fe526316850d7669a7fd1e90f54b0aefacbe636f2d29bdf44cR77) [[3]](diffhunk://#diff-107b5c09484457920310077d4839b0b33af4705e144d0f4a436176e5aae8722fR28) [[4]](diffhunk://#diff-81ea5a1a1cc042b72c3a3e3a0cb02b0b12a2262df1f474f8f2664ecf6e643fd1R153)
* Introduced new modules for ScyllaDB connection pooling and table provider factories: `scylladb.rs` in `data_components`, `scylladbpool.rs` and `scylladbconn.rs` in `db_connection_pool`, and corresponding mod entries guarded by the `scylladb` feature flag. [[1]](diffhunk://#diff-0a7c082bd55feb73df2f690d81f761b6d7fc6f083834400fbe32eb3a95b7ce8bR60-R61) [[2]](diffhunk://#diff-80c51dc22d51d556bb061a3308ac6aa7f377beb0d2e981e6ba5f5fd8b76c5d38R21-R22) [[3]](diffhunk://#diff-634890ca28135f2e76361ea39b6a6ca49925fb90b35c692763d34db4079f8678L22-R23) [[4]](diffhunk://#diff-971a096d0e80a49286b36a9e456d092f60054eed16053ed161c865608ce3b9f4R1-R230) [[5]](diffhunk://#diff-a1bcd2a31ca5e9e6f1685df66eeb64dbd9ca7c58005aa6fe7b73184c916ae1fdR1-R193) [[6]](diffhunk://#diff-eff06eaa7ca034e1e520d9cfd7b53cd8dd7ba47ef3b3eb5cac534a8e987d547bR180-R181)

**Feature Flag and Dependency Grouping**

* Defined the `scylladb` feature and grouped dependencies for ScyllaDB support in `Cargo.toml` files, enabling conditional compilation and modular integration. [[1]](diffhunk://#diff-bcbe44138259c7fe526316850d7669a7fd1e90f54b0aefacbe636f2d29bdf44cR147) [[2]](diffhunk://#diff-107b5c09484457920310077d4839b0b33af4705e144d0f4a436176e5aae8722fR67) [[3]](diffhunk://#diff-81ea5a1a1cc042b72c3a3e3a0cb02b0b12a2262df1f474f8f2664ecf6e643fd1R285-R289)

**Testing and Validation**

* Added comprehensive unit tests for ScyllaDB connection pool and table provider modules to validate error handling, trait implementations, and edge cases. [[1]](diffhunk://#diff-971a096d0e80a49286b36a9e456d092f60054eed16053ed161c865608ce3b9f4R1-R230) [[2]](diffhunk://#diff-a1bcd2a31ca5e9e6f1685df66eeb64dbd9ca7c58005aa6fe7b73184c916ae1fdR1-R193)

**Documentation and Example Configuration**

* Added a new federated Spicepod YAML example for ScyllaDB under `test/spicepods/tpch/sf1/federated/scylladb.yaml`, demonstrating how to configure datasets from ScyllaDB with parameter anchors.

### CODE REVIEW

File: test/spicepods/tpch/sf1/federated/dynamodb[scylladb-alternator].yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# Define common anchors
+definitions:
+  # Common ScyllaDB Alternator parameters
+  - &scylladb_alternator_params # Use a dummy region (required by AWS SDK but not used by Alternator)
+    dynamodb_aws_region: us-east-1
+    # Alternator doesn't require AWS credentials by default
+    # Set aws_auth to 'key' with dummy values or configure Alternator auth
+    dynamodb_aws_auth: key
+    dynamodb_aws_access_key_id: ${env:SCYLLADB_ALTERNATOR_ACCESS_KEY_ID:none}
+    dynamodb_aws_secret_access_key: ${env:SCYLLADB_ALTERNATOR_SECRET_ACCESS_KEY:none}
```
Comment (phillipleblanc) [inline test/spicepods/tpch/sf1/federated/dynamodb[scylladb-alternator].yaml]: We don't support this secret syntax of having a default

File: crates/runtime/src/dataconnector/scylladb.rs [hunk 2]
```diff
@@ -0,0 +1,615 @@
...
+        .description("The username to use to authenticate with ScyllaDB."),
+    // scylladb_pass
+    ParameterSpec::component("pass")
+        .secret()
+        .description("The password to use to authenticate with ScyllaDB."),
+    // scylladb_datacenter
+    ParameterSpec::component("datacenter")
+        .description("The datacenter to use for local connection preferences."),
+    // scylladb_ssl
+    ParameterSpec::component("ssl")
```
Comment (phillipleblanc) [inline crates/runtime/src/dataconnector/scylladb.rs:118]: This should default to true to be in line with our secure by default principle

Comment (phillipleblanc) [reply crates/runtime/src/dataconnector/scylladb.rs:118]: I would also expect configuration to be able to provide the CA cert to validate the ScyllaDB certificate against?

File: crates/db_connection_pool/src/dbconnection/scylladbconn.rs [hunk 3]
```diff
@@ -0,0 +1,1791 @@
...
+        let mut typed_stream =
+            rows_iter
+                .rows_stream::<scylla::value::Row>()
+                .map_err(|e| Error::ConversionError {
+                    message: format!("Type check failed: {e}"),
+                })?;
+
+        // Note: Using stream! macro here is necessary for async iteration.
+        // The AGENTS.md guidance to avoid stream! is acknowledged but this is
+        // unavoidable for converting async iterators to Arrow streams.
```
Comment (phillipleblanc) [inline crates/db_connection_pool/src/dbconnection/scylladbconn.rs:257]: This is false - avoiding `stream!` is entirely possible - but I don't think it matters too much here.

**Review by phillipleblanc — DISMISSED** [review]

**Review by sgrebnov — DISMISSED** [review]

**Review by phillipleblanc — APPROVED** [review]
